docs/development/awk/intro-to-awk/intro-to-awk.md

```diff
@@ -0,0 +1,732 @@
+New file: givenLine.awk
+New file: doesNotExist
+Cannot read doesNotExist
+Total number of files processed: 3
+{{< /output >}}
+
+## Practical examples
+
+In this section of the guide you will see practical uses of AWK. The best advice that I
+can give you is to try all these examples on your own UNIX machines.
+
+### Printing a Given Line from a File
+
+Imagine that you want to print a given line from a text file. Well, AWK can help
+you. The AWK code that implements that is the following:
+
+{{< file "givenLine.awk" awk >}}
+{
+	if (NR==line)
+	{
+		print $0;
+	}
+}
+{{< /file >}}
+
+The `givenLine.awk` uses a parameter named `line` that should be given as a command
+line argument to the AWK script using the `-v` flag.
+
+Executing `givenLine.awk` will generate the following kind of output:
+
+    awk -v line=1 -f givenLine.awk ./givenLine.awk
+{{< output >}}
+{
+{{< /output >}}
+
+    awk -v line=4 -f givenLine.awk ./givenLine.awk
+{{< output >}}
+		print $0;
+{{< /output >}}
+
+### Printing Lines that Follow a Numeric Pattern
+
+In this section you will learn how to print lines 10, 20, 30, etc. of a file using
+a variation of the code found in `givenLine.awk`. The AWK code is saved as
+`multipleLines.awk` and is as follows:
+
+{{< file "multipleLines.awk" awk >}}
+{
+	if (NR % line == 0)
+	{
+		print $0;
+	}
+}
+{{< /file >}}
+
+The trick here is done by the modulus operator (`%`).
+
+Executing `multipleLines.awk` will generate the following kind of output:
+
+    awk -v line=1 -f multipleLines.awk ./multipleLines.awk
+{{< output >}}
+{
+	if (NR % line == 0)
+	{
+		print $0;
+	}
+}
+{{< /output >}}
+
+If the value of `line` is `1`, then `multipleLines.awk` will print the contents of
+all input files.
+
+### Printing two Given Fields from a File
+
+Let us say that you have a file with multiple fields and that you want to print the
+values of the 1st and the 3rd fields only. Well, AWK can help you with this as follows:
+
+{{< file "field1and3.awk" awk >}}
+{
+	print $1, $3;
+}
+{{< /file >}}
+
+Executing `field1and3.awk` will generate the following kind of output:
+
+    awk -f field1and3.awk field1and3.awk
+{{< output >}}
+{
+print $3;
+}
+{{< /output >}}
+
+As `field1and3.awk` is pretty small, you can also execute it as follows:
+
+    awk '{print $1, $3}' field1and3.awk
+{{< output >}}
+{
+print $3;
+}
+{{< /output >}}
+
+### Counting Lines
+
+In this subsection you will learn how to count the number of lines that were given to AWK
+for processing.
+
+{{< file "countLines.awk" awk >}}
+{
+	if (FNR==1) {
+		print "Processing:", FILENAME;
+	}
+}
+
+END {
+	print "Read", NR, "records in total";
+}
+{{< /file >}}
+
+The use of `FNR` makes sure that the filename of each processed file will be
+printed only once whereas the use of `END` makes sure that the results will be
+printed just before the AWK finished its job.
+
+Executing countLines.awk will generate the following kind of output:
+
+	awk -f countLines.awk countLines.awk
```

Comment:
Can we avoid feeding the script into itself? We'd like to have these examples restructured so that they all work off the same static text file, if possible

Reply:
I do not see any problems with this.

Reply:
I should have clarified myself here earlier--I don't mean to call into question the technical correctness of the examples as you've written them. Instead, we view passing an awk script as an argument to an awk script as potentially confusing to the reader:

- When two awk scripts (or more) are passed as arguments to the awk command, it could be less clear which is serving as the executed script and which is the file that is being operated on.
- Part of the work of learning a language like AWK is interpreting the syntax and forming a mental model of how it works. When processing a script with another AWK script, we're asking the reader to analyze the file again as AWK would; instead of parsing the meaning of the file, they will have to perform a literal parsing of the characters in the file.

It's definitely possible to read through the examples as they are now, but we think these issues can be hurdles that complicate the learning process. Wherever we can, we prefer to isolate a new concept that is being introduced so that it is easier to digest. For this reason, we ask that your examples process a file separate from the script itself, as with the [Calculating Word Frequencies example](https://deploy-preview-2561--nostalgic-ptolemy-b01ab8.netlify.com/development/awk/intro-to-awk/intro-to-awk/#calculating-word-frequencies).

I want to take a moment to say that we think that having these practical examples are really wonderful, and you've inspired us to include similar sections in other guides that we're writing. 

If you have any disagreement with these notes, please let us know! We appreciate your working with us.